spectrumx_visualization_platform/spx_vis/api/views.py

```diff
@@ -1,119 +1,212 @@
+
+    for capture in captures_data:
+        timestamp = capture.get("timestamp", "")
+        scan_time = capture.get("scan_time", 0.08121538162231445)  # Default value
+        end_time = calculate_end_time(timestamp, scan_time)
+
+        print(f"Local Capture Debug: timestamp={timestamp}, scan_time={scan_time}, computed end_time={end_time}")
+
+        capture_dict = {
+            "id": capture.get("_id", capture.get("id", "unknown")),
+            "name": capture["name"],
+            "media_type": capture.get("metadata", {}).get("data_type", "unknown"),
+            "timestamp": timestamp,
+            "created_at": timestamp,
+            "source": "svi_user",
+            "files": capture["files"],
+            "owner": request.user.id,
+            "type": capture.get("metadata", {}).get("data_type", "rh"),
+            "min_freq": capture.get("metadata", {}).get("fmin", ""),
+            "max_freq": capture.get("metadata", {}).get("fmax", ""),
+            "scan_time": scan_time,
+            "end_time": end_time,
+        }
+        file_list.append(capture_dict)
+
+    # Combine lists
+    combined_file_list = sds_file_list + file_list
+
+    # filtering parameters
+    min_frequency = request.query_params.get("min_frequency")
+    max_frequency = request.query_params.get("max_frequency")
+    start_time = datetime_check(request.query_params.get("start_time"))
+    end_time = datetime_check(request.query_params.get("end_time"))
+    source_filter = request.query_params.get("source")
+
+    print(f"source val:", source_filter)
+
+    print(f"min_freq={min_frequency}, max_freq={max_frequency}, start_time={start_time}, end_time={end_time}")
+
+    if min_frequency or max_frequency or start_time or end_time or source_filter:
+        min_freq = float(min_frequency) if min_frequency else None
+        max_freq = float(max_frequency) if max_frequency else None
+
+        def filter_capture(capture):
+            capture_min_freq = float_check(capture.get("min_freq"))
+            capture_max_freq = float_check(capture.get("max_freq"))
+            capture_start_time = datetime_check(capture.get("timestamp"))
+            capture_end_time = datetime_check(capture.get("end_time"))
+
+            if min_freq is not None and capture_max_freq < min_freq:
+                return False
+            if max_freq is not None and capture_min_freq > max_freq:
+                return False
+            if start_time and (capture_end_time is None or capture_end_time < start_time):
+                return False
+            if end_time and (capture_start_time is None or capture_start_time > end_time):
+                return False
+
+            if source_filter and capture.get("source") != source_filter:
+                return False
+
+            return True
+
+        combined_file_list = list(filter(filter_capture, combined_file_list))
+
+    return Response({"captures": combined_file_list, "sds_count": sds_file_count})
```

Comment:
I don't think there's a need to return a separate "sds_count" field because we can figure that out by looking at the "source" fields on the captures themselves. Returning more than just the captures complicates the API and the frontend logic in turn